### PR TITLE
UCP/CORE: Updated methods working with request ptr map.

### PR DESCRIPTION
**Pull request by rakhmets**
## What
Removed extra method `ucp_ep_ptr_id_alloc`.
Re-named `ucp_ep_ptr_map_check_status` to `ucp_request_ptr_map_status_check`.

## Why ?
Addressed https://github.com/openucx/ucx/pull/7159#discussion_r689850233

### CODE REVIEW

**Review by yosefe — APPROVED** [review]

**Comment by yosefe** [comment]
@evgeny-leksikov can you pls take a look as well?

File: src/ucp/core/ucp_request.inl [hunk 1]
```diff
@@ -852,33 +852,24 @@ ucp_request_get_memory_type(ucp_context_h context, const void *address,
 }
 
 static UCS_F_ALWAYS_INLINE void
-ucp_ep_ptr_map_check_status(ucp_ep_h ep, void *ptr, const char *action_str,
-                            ucs_status_t status)
+ucp_request_ptr_map_check_status(ucp_ep_h ep, void *ptr, const char *action_str,
```
Comment (evgeny-leksikov) [inline src/ucp/core/ucp_request.inl:855]: name looks wrong

File: src/ucp/core/ucp_request.inl [hunk 2]
```diff
@@ -852,33 +852,24 @@ ucp_request_get_memory_type(ucp_context_h context, const void *address,
...
 static UCS_F_ALWAYS_INLINE void ucp_send_request_id_alloc(ucp_request_t *req)
 {
     ucp_ep_h ep = req->send.ep;
     ucs_status_t status;
 
     ucp_request_id_check(req, ==, UCS_PTR_MAP_KEY_INVALID);
-    status = ucp_ep_ptr_id_alloc(ep, req, &req->id);
+    status = UCS_PTR_MAP_PUT(request, &ep->worker->request_map, req,
+                             ucp_ep_use_indirect_id(ep), &req->id);
+    ucp_request_ptr_map_check_status(ep, req, "allocate", status);
```
Comment (evgeny-leksikov) [inline src/ucp/core/ucp_request.inl:871]:
```suggestion
    ucp_request_ptr_map_status_check(status, "PUT", ep, req);
```
or even maybe add a helper macro like `UCP_REQUEST_PTRMAP_PUT(&ep->worker->request_map, req)` with put+check inside?

Comment (rakhmets) [reply src/ucp/core/ucp_request.inl]: Applied suggested changes.

Macro looks redundant for me, cause it doesn't simplify the code too much to be introduced, IMO.

**Review by evgeny-leksikov — APPROVED** [review]

**Review by yosefe — APPROVED** [review]

**Pull request merged**